### PR TITLE
fix for 360+ degrees wedgebuffer

### PR DESCRIPTION
**Pull request by raymondnijssen**
## Description
Geometry function createWedgeBuffer() now handles the case of angularWidth >= 360, which should return either a full circle geometry or a donut geometry in case there is an innerRadius.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Review by nyalldawson — APPROVED** [review]
Looks good! Can you add a unit test covering this too?

**Comment by raymondnijssen** [comment]
@nyalldawson Tnx! I will try adding a test later today.

File: src/core/geometry/qgsgeometry.cpp [hunk 1]
```diff
@@ -260,6 +260,47 @@ QgsGeometry QgsGeometry::collectGeometry( const QVector< QgsGeometry > &geometri
 
 QgsGeometry QgsGeometry::createWedgeBuffer( const QgsPoint &center, const double azimuth, const double angularWidth, const double outerRadius, const double innerRadius )
 {
+  if ( angularWidth >= 360.0 )
+  {
+    std::unique_ptr< QgsCompoundCurve > outerCc = qgis::make_unique< QgsCompoundCurve >();
```
Comment (nyalldawson) [inline src/core/geometry/qgsgeometry.cpp]: Have you looked into the QgsCircle class? That might be a better way to construct these. There's a constructor which takes a center point and radius, and then you can call toCircularString() to create the curve for the exterior/interior rings.

**Comment by nyalldawson** [comment]
@raymondnijssen 

For the test you'll need to add cases with > 360 to the existing tests at https://github.com/qgis/QGIS/blob/master/tests/src/python/test_qgsgeometry.py#L4359

Thanks!

File: src/core/geometry/qgsgeometry.cpp (hunk 1 above)
Comment (raymondnijssen) [reply src/core/geometry/qgsgeometry.cpp]: Done

**Comment by m-kuhn** [comment]
Wow, really great looking!

**Comment by nyalldawson** [comment]
Thank you!

**Pull request merged**